backend/src/main/java/fatec/jpa/eventoapp/controllers/EventController.java

```diff
@@ -10,27 +11,43 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import fatec.jpa.eventoapp.entity.PublicEvent;
+import fatec.jpa.eventoapp.entity.User;
 import fatec.jpa.eventoapp.exception.BadRequestException;
 import fatec.jpa.eventoapp.exception.NotFoundException;
 
 @WebServlet("/events")
-public class EventController extends HttpServlet {
+public class  EventController extends BaseController implements BaseControllerInterface {
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
         EntityManager manager = PersistenceManager.getInstance().getEntityManager();
+        ObjectMapper objectMapper = new ObjectMapper();
+        String result = "";
+
         String idParam = req.getParameter("id");
+        String usernameParam = req.getParameter("username");
 
-        if (idParam == null) throw new BadRequestException("Id param not found");
+        if (idParam != null) {
+            EventDaoJpa eventDaoJpa = new EventDaoJpa(manager);
+            Event event = eventDaoJpa.findById(Integer.parseInt(idParam), true);
 
-        ObjectMapper objectMapper = new ObjectMapper();
-        EventDaoJpa eventDaoJpa = new EventDaoJpa(manager);
-        Event event = eventDaoJpa.findById(Integer.parseInt(idParam));
+            if (event == null) throw new NotFoundException("Event not found");
```

Comment:
Ao invés de lançar um throw, não poderíamos criar uma classe Response com o atributo message, e lançar HttpStatus 400 (bad request) com uma message de Event not found?

Reply:
Essa exceção cai no [exception filter](https://github.com/NicolasFujisawa/fatec-jpa-eventoapp/blob/front-end-app/backend/src/main/java/fatec/jpa/eventoapp/filter/ExceptionHandlerFilter.java#L34) , que faz justamente isso